easybuild/tools/singularity/utilities.py

```diff
@@ -0,0 +1,395 @@
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+All required to provide details of build environment 
+and allow for reproducable builds
+
+:author: Shahzeb Siddiqui (Pfizer)
+"""
+import subprocess
+import os
+import sys
+import easybuild.tools.options as eboptions
+
+from vsc.utils import fancylogger
+from easybuild.tools.config import build_option, get_module_naming_scheme, singularity_path
+from easybuild.tools.filetools import change_dir, which, write_file
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.testing import  session_state
+_log = fancylogger.getLogger('tools.package')  # pylint: disable=C0103
+
+def architecture_query(model_num):
+	model_mapping = {
+		'4F': 'Broadwell',
+		'57': 'KnightsLanding',
+		'3F': 'Haswell',
+		'46': 'Haswell',
+		'3A': 'IvyBridge',
+		'3E': 'IvyBridge',
+		'2A': 'SandyBridge',
+		'2D': 'SandyBridge',
+		'25': 'Westmere',
+		'2C': 'Westmere',
+		'2F': 'Westmere',
+		'1E': 'Nehalem',
+		'1A': 'Nehalem',
+		'2E': 'Nehalem',
+		'17': 'Penryn',
+		'1D': 'Penryn',
+		'0F': 'Merom'
+		}
+	if model_num in model_mapping.keys():
+		return model_mapping[model_num]
+	else:
+		print "Model Number: ", model_num, " not found in dictionary, please consider adding the model number and Architecture name"
+		return None
+
+
+def generate_singularity_recipe(ordered_ecs,bootstrap_opts,easyconfig_repo,easyblock_repo):
+
+    image_name = build_option('imagename')
+    image_format = build_option('imageformat')
+    build_image = build_option('buildimage')
+    import_ec_repo = easyconfig_repo
+    sing_path = singularity_path()
+    ec_repo = ""
+    eb_repo = ""
+    ec_branch = ""
+    eb_branch = ""
+    easyblock_file = ""
+
+
+    # check if --singularitypath is valid path and a directory
+    if os.path.exists(sing_path) and os.path.isdir(sing_path):
+	singularity_writepath = singularity_path()
+    else:
+	msg = "Invalid path: " +  sing_path +  " please specify a valid directory path"
+	print msg
+	EasyBuildError(msg)
```

Comment:
You should either raise this exception or `sys.exit` with a non zero value to flag an error.